multiqc/modules/samtools/rmdup.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from Samtools rmdup """
+
+import logging
+import re
+from collections import OrderedDict, defaultdict
+from multiqc import config, plots
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class RmdupReportMixin():
+
+    def parse_samtools_rmdup(self):
+        """ Find Samtools rmdup logs and parse their data """
+
+        self.samtools_rmdup = dict()
+        for f in self.find_log_files(config.sp['samtools']['rmdup']):
+            parsed_data = parse_single_report(f['f'])
+            if len(parsed_data) > 0:
+                if f['s_name'] in self.samtools_rmdup:
+                    log.debug("Duplicate sample name found! Overwriting: {}".format(f['s_name']))
+                self.add_data_source(f, section='rmdup')
+                self.samtools_rmdup[f['s_name']] = parsed_data
+
+        if len(self.samtools_rmdup) > 0:
+
+            # Write parsed report data to a file (restructure first)
+            self.write_data_file(self.samtools_rmdup, 'multiqc_samtools_rmdup')
+
+        return len(self.samtools_idxstats)
+
+def parse_single_report(f):
+    """ Parse a samtools rmdup """
+
+    """ Go through log file looking for rmdup output.
+    If the
+    Grab the name from the RG tag of the preceding bwa command """
+
+    # Example below:
+    # [bam_rmdupse_core] 26602816 / 103563641 = 0.2569 in library '   '
+        
+    dups_regex = "\[bam_\w+_core\] (\d+) / (\d+) = \((\d+.\d+)\) in library"
+    # dups_regex = "rmdup: (Removed|Marked) (\d+) of (\d+) \((\d+.\d+)%\) read ids as duplicates"
+    input_file_regex = "rmdup: Opening (\S+) for read."
+    # rgtag_name_regex = "\\\\tID:(\S*?)\\\\t"
+    parsed_data = {}
+    s_name = None
+    for l in f.splitlines():
+        # try to find name from RG-tag. If bwa mem is used upstream rmdup with pipes, then the bwa mem command
+        # including the read group will be written in the log
+
+        # match = re.search(rgtag_name_regex, l)
+        # if match:
+        #     s_name = match.group(1)
+
+        # try to find name from the input file name, if used
+        match = re.search(input_file_regex, l)
```

Comment:
All of your code in this function to find the sample name from the log is not used, as the function doesn't do anything with it - it's not returned.

I think what's happening here is that you're mixing chunks of copied code from multiple modules which have different strategies. If you're planning to try to set the sample name from the file contents you probably want to add the parsed data directly to the main dictionary in this function, rather than returning it as a dict.